File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/PromotionLevel.java
package net.nemerosa.ontrack.model;

import lombok.Data;

@Data
public class PromotionLevel implements Comparable<PromotionLevel> {

    private final String id;
    private final String name;
    private final int order;
    private final String description;

    @Override
    public int compareTo(PromotionLevel o) {
        return this.order - o.order;
    }
}

File: ontrack-kdsl/src/main/java/net/nemerosa/ontrack/kdsl/spec/PromotionLevel.kt
package net.nemerosa.ontrack.kdsl.spec

import net.nemerosa.ontrack.kdsl.connector.Connector
import net.nemerosa.ontrack.kdsl.connector.graphql.schema.type.ProjectEntityType

/**
 * Representation of a promotion level.
 *
 * @property connector Ontrack connector
 * @property id Promotion level ID
 * @property name Promotion level name
 * @property description Promotion level description
 */
class PromotionLevel(
    connector: Connector,
    id: UInt,
    val name: String,
    val description: String?,
) : ProjectEntity(connector, ProjectEntityType.PROMOTION_LEVEL, id)
